### PR TITLE
feat(output): Add HTML outputs to Prowler

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

HTML outputs are back for all the providers.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by MrCloudSec** [comment]
Great job!! Please change the screenshot in the docs index and solve the kubernetes summary tables.

File: docs/tutorials/reporting.md [hunk 1]
```diff
@@ -294,14 +294,16 @@ The following code is an example output of the [JSON-ASFF](https://docs.aws.amaz
 ???+ note
     Each finding is a `json` object within a list.
 
+### HTML
+
+The following photo is an example of the HTML output:
```
Comment (jfagoagas) [inline docs/tutorials/reporting.md]:
```suggestion
The following image is an example of the HTML output:
```

File: prowler/__main__.py [hunk 2]
```diff
@@ -269,6 +270,17 @@ def prowler():
                     mode,
                 )
 
+            if "html" in mode:
+                add_html_footer(
+                    global_provider.output_options.output_filename,
+                    args.output_directory,
```
Comment (jfagoagas) [inline prowler/__main__.py]:
```suggestion
                    global_provider.output_options.output_directory,
```

File: prowler/__main__.py [hunk 3]
```diff
@@ -269,6 +270,17 @@ def prowler():
...
+                )
+                fill_html_overview_statistics(
+                    stats,
+                    global_provider.output_options.output_filename,
+                    args.output_directory,
```
Comment (jfagoagas) [inline prowler/__main__.py]:
```suggestion
                    global_provider.output_options.output_directory,
```

File: prowler/__main__.py (hunk 2 above)
Comment (jfagoagas) [reply prowler/__main__.py]: Please modify this too in line 269.

File: prowler/lib/cli/parser.py [hunk 4]
```diff
@@ -146,8 +146,8 @@ def __init_outputs_parser__(self):
             "-M",
             nargs="+",
             help="Output modes, by default csv and json-oscf are saved. When using AWS Security Hub integration, json-asff output is also saved.",
-            default=["csv", "json-ocsf"],
-            choices=["csv", "json-asff", "json-ocsf"],
+            default=["csv", "json-ocsf", "html"],
```
Comment (jfagoagas) [inline prowler/lib/cli/parser.py:149]: Please fix the tests related with this change.

File: prowler/lib/outputs/compliance/compliance.py [hunk 5]
```diff
@@ -203,12 +204,14 @@ def get_check_compliance(finding, provider_type, output_options) -> dict:
     }
     """
     try:
+        if not isinstance(finding, FindingOutput):
+            check_id = finding.check_metadata.CheckID
+        else:
+            check_id = finding.check_id
```
Comment (jfagoagas) [inline prowler/lib/outputs/compliance/compliance.py]: Why this change? The HTML should get this from the finding object.

File: prowler/lib/outputs/html/html.py [hunk 6]
```diff
@@ -0,0 +1,567 @@
...
+
+from prowler.config.config import (
+    html_file_suffix,
+    html_logo_url,
+    prowler_version,
+    square_logo_img,
+    timestamp,
+)
+from prowler.lib.logger import logger
+from prowler.lib.outputs.compliance.compliance import get_check_compliance
```
Comment (jfagoagas) [inline prowler/lib/outputs/html/html.py]: This shouldn't be needed here, please get that information from the finding object.

File: prowler/lib/outputs/html/html.py [hunk 7]
```diff
@@ -0,0 +1,567 @@
...
+
+def add_html_header(file_descriptor, provider):
+    try:
+        parameters = sys.argv[1:]
+        for index, parameter in enumerate(parameters):
+            if (
+                parameter == "--kubeconfig-file"
+                and "/.kube/config" in parameters[index + 1]
+            ):
+                parameters[index + 1] = "~/.kube/config"
```
Comment (jfagoagas) [inline prowler/lib/outputs/html/html.py]: Why this? Please leave this function jus to fill the header, if you need to do some other changes pass do them in another place.

File: prowler/lib/outputs/html/html.py [hunk 8]
```diff
@@ -0,0 +1,567 @@
...
+def get_assessment_summary(provider):
+    """
+    get_assessment_summary gets the HTML assessment summary for the provider
+    """
+    try:
+        # This is based in the Provider_provider class
+        # It is not pretty but useful
+        # AWS_provider --> aws
+        # GCP_provider --> gcp
+        # Azure_provider --> azure
```
Comment (jfagoagas) [inline prowler/lib/outputs/html/html.py:509]: I think we need to do it for K8s right?

File: prowler/lib/outputs/outputs.py [hunk 9]
```diff
@@ -139,6 +140,12 @@ def report(check_findings, provider):
                             )
                             file_descriptors["json-ocsf"].write(",")
 
+                        if "html" in file_descriptors:
+                            fill_html(
+                                file_descriptors["html"], finding_output, output_options
+                            )
+                            file_descriptors["html"].write("")
```
Comment (jfagoagas) [inline prowler/lib/outputs/outputs.py]: Why is this needed?

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Great work bringing back the HTML format @pedrooot 👏 

I left some comments since I don't understand some parts of the code. We need to make some parts clearer and fix the failing tests.

File: prowler/lib/outputs/html/html.py (hunk 8 above)
Comment (pedrooot) [reply prowler/lib/outputs/html/html.py:509]: Forgot to update the comment but it's already done for Kubernetes

File: docs/tutorials/reporting.md (hunk 1 above)
Comment (pedrooot) [reply docs/tutorials/reporting.md]: Done

File: prowler/__main__.py (hunk 2 above)
Comment (pedrooot) [reply prowler/__main__.py]: Done

File: prowler/__main__.py (hunk 3 above)
Comment (pedrooot) [reply prowler/__main__.py]: Done

File: prowler/lib/outputs/outputs.py (hunk 9 above)
Comment (pedrooot) [reply prowler/lib/outputs/outputs.py]: When the report is written, first of all the html header is created and the method fill_html is called when creating the report.

Comment (jfagoagas) [reply prowler/lib/outputs/outputs.py]: I know but I mean about ` file_descriptors["html"].write("")`

File: prowler/lib/cli/parser.py (hunk 4 above)
Comment (pedrooot) [reply prowler/lib/cli/parser.py:149]: Done

File: prowler/lib/outputs/html/html.py (hunk 6 above)
Comment (pedrooot) [reply prowler/lib/outputs/html/html.py]: True

File: prowler/lib/outputs/compliance/compliance.py (hunk 5 above)
Comment (pedrooot) [reply prowler/lib/outputs/compliance/compliance.py]: True

File: prowler/lib/outputs/html/html.py (hunk 7 above)
Comment (pedrooot) [reply prowler/lib/outputs/html/html.py]: Done

File: prowler/lib/outputs/outputs.py (hunk 9 above)
Comment (pedrooot) [reply prowler/lib/outputs/outputs.py]: Fact, it's unneeded

**Review by jfagoagas — APPROVED** [review]
Great work @pedrooot 👏 HTML is back 🚀
